core/src/test/java/kafka/server/share/SharePartitionTest.java

```diff
@@ -1065,6 +1068,727 @@ public void testAcquireReleasedRecordMultipleBatches() {
+        assertNull(sharePartition.cachedState().get(10L).offsetState().get(17L).acquisitionLockTimeoutTask());
+        assertEquals(3, sharePartition.timer().size());
+
+        // Allowing acquisition lock to expire.
+        TestUtils.waitForCondition(
+                () -> {
+                    Map<Long, InFlightState> expectedOffsetStateMap = new HashMap<>();
+                    expectedOffsetStateMap.put(10L, new InFlightState(RecordState.AVAILABLE, (short) 1, EMPTY_MEMBER_ID));
+                    expectedOffsetStateMap.put(11L, new InFlightState(RecordState.AVAILABLE, (short) 1, EMPTY_MEMBER_ID));
+                    expectedOffsetStateMap.put(12L, new InFlightState(RecordState.AVAILABLE, (short) 2, EMPTY_MEMBER_ID));
+                    expectedOffsetStateMap.put(13L, new InFlightState(RecordState.AVAILABLE, (short) 2, EMPTY_MEMBER_ID));
+                    expectedOffsetStateMap.put(14L, new InFlightState(RecordState.AVAILABLE, (short) 2, EMPTY_MEMBER_ID));
+                    expectedOffsetStateMap.put(15L, new InFlightState(RecordState.AVAILABLE, (short) 1, EMPTY_MEMBER_ID));
+                    expectedOffsetStateMap.put(16L, new InFlightState(RecordState.AVAILABLE, (short) 1, EMPTY_MEMBER_ID));
+                    expectedOffsetStateMap.put(17L, new InFlightState(RecordState.AVAILABLE, (short) 1, EMPTY_MEMBER_ID));
+
+                    return sharePartition.timer().size() == 0 &&
+                            sharePartition.nextFetchOffset() == 10 &&
+                            expectedOffsetStateMap.equals(sharePartition.cachedState().get(10L).offsetState());
+                },
+                DEFAULT_MAX_WAIT_ACQUISITION_LOCK_TIMEOUT_MS,
+                () -> "Acquisition lock never got released.");
+        assertNull(sharePartition.cachedState().get(10L).offsetState().get(10L).acquisitionLockTimeoutTask());
+        assertNull(sharePartition.cachedState().get(10L).offsetState().get(11L).acquisitionLockTimeoutTask());
+        assertNull(sharePartition.cachedState().get(10L).offsetState().get(12L).acquisitionLockTimeoutTask());
+        assertNull(sharePartition.cachedState().get(10L).offsetState().get(13L).acquisitionLockTimeoutTask());
+        assertNull(sharePartition.cachedState().get(10L).offsetState().get(14L).acquisitionLockTimeoutTask());
+        assertNull(sharePartition.cachedState().get(10L).offsetState().get(15L).acquisitionLockTimeoutTask());
+        assertNull(sharePartition.cachedState().get(10L).offsetState().get(16L).acquisitionLockTimeoutTask());
+        assertNull(sharePartition.cachedState().get(10L).offsetState().get(17L).acquisitionLockTimeoutTask());
+    }
+
+    @Test
+    public void testAcquisitionLockOnAcknowledgingMultipleSubsetRecordBatchWithGapOffsets() throws InterruptedException {
+        SharePartition sharePartition = SharePartitionBuilder.builder().withAcquisitionLockTimeoutMs(ACQUISITION_LOCK_TIMEOUT_MS).build();
```

Comment:
I appreciate the level of unit testing here, but some of the tests take quite a lot of reading to see what they're doing. This one in particular could do with a comment explaining what you're trying to achieve in terms of offsets and states. The comment "Untracked gap of 3 offsets from 7-9." took quite a bit of reading to see what was meant.

Reply:
yeah, i should probably have records as variables in such cases rather than just supplying them directly in acquire method, that would make these kind of tests more readable.